### DISCUSSION TITLE
Way to club webparts as Tabs or Accordions

### DISCUSSION THREAD

**Discussion by spverma03** [pnp/sp-dev-fx-webparts#2109] [Q&A] [answered]
### Sample description

Is there any way to load OOB or custom web parts as Tabs or Accordions? The use case here is to show multiple webparts in one view so that to save space on the page. For example, configure multiple news webparts for different department sites. Then club those department news webparts in one webpart as tabs and show on organization portal.

It was possible in classic pages and maybe in old SPFx versions, but is there any recommended way in Modern Pages?

### Are you willing to help?

Yes

**Comment by ghost** [comment]
Thank you for reporting this issue. We will be triaging your incoming issue as soon as possible.

**Comment by Abderahman88** [comment]
Hi @spverma03,

Yes, it's possible. You can use [@pnp/sp-dev-fx-controls-react controls](https://pnp.github.io/sp-dev-fx-controls-react/controls/Accordion.html) for example.

![image](https://user-images.githubusercontent.com/36161889/126637558-a91702d2-1d25-49ed-99c8-dcbba692af80.png)

Another possibility, is to put `Quick Links` on the portal and redirect to the sitecollection of each department with their news.

![image](https://user-images.githubusercontent.com/36161889/126638164-6e278dda-5b3e-4c6b-820e-611bd9574bc6.png)

**Comment by spverma03** [comment]
@Abderahman88 

Thank you for your quick response. These do not fulfill requirements. We need a way to load SP Modern Web-parts inside tabs/accordion. 

Also, the second option (Quick Links) is not relevant as the primary requirement is to show news at one place instead of re-directing users to different page/site.

Something similar to this but in modern it seems not possible: https://www.paitgroup.com/blog/sharepoint-tabbed-web-partshillbillytabs-3-0

**Comment by Abderahman88** [comment]
@spverma03 Standard webparts can not be loaded inside tabs/accordion. You need to create a custom news webpart.

Update: new feature - collapsible sections

**Comment by PathToSharePoint** [comment]
Modern SharePoint recently added collapsible sections, which should address your requirement to "save space on the page.".

I am the author of the original "Easy Tabs Web Part" for classic pages, and AFAIK nothing equivalent has been done in modern.

**Comment by Abderahman88** [comment, marked as answer]
@spverma03 Collapsible sections is in roll out phase. Maybe you can already use it.

![image](https://user-images.githubusercontent.com/36161889/126692178-4f492a20-cd3e-4aa7-b62c-5d1604bcbc0d.png)

An example is as below (this is without the collapsible sections, because it's not active on my tenant)

![image](https://user-images.githubusercontent.com/36161889/126691918-1fcb982e-adbb-4c31-a16a-e3232d78d94e.png)

**Comment by spverma03** [comment]
So what I understand is:

**1. Accordion** - Collapsible sections can be used, but it will take **full width on page.** We cannot make accordion for specific zones.

**2. Tabs** - This is not possible currently and we need to create custom webpart. The pain here would be to create property-pane. OOB webparts are having nice property pane settings and not all code/API etc are exposed for customizations. It will be difficult to provide features same as OOB webpart inside a custom webpart. The OOB webparts set a high expectation to end-users and they demand similar experience in custom webparts as well. 

_I would appreciate if someone can help with some sort of compatibility with Tabs as well._

**Query:** The roadmap mentions **tabs** as well. _Is it anyway going to come because I can see only Accordion view in the demo videos. I do not have this feature yet in my tenant._

SharePoint: Collapsible sections on modern pages

You’ll have the ability to show page sections in an accordion view (collapsed or expanded) **or as tabs**. The accordion view will be collapsed by default, but can be set to show expanded.

**Reply by hugoabernier** [reply]
Converted this topic to a conversation. Let's keep this topic going!
